dataset/src/main/scala/frameless/functions/AggregateFunctions.scala

```diff
@@ -23,71 +23,46 @@ trait AggregateFunctions {
   def sum[A, T, Out](column: TypedColumn[T, A])(
     implicit
     summable: CatalystSummable[A, Out],
-    encoder1: TypedEncoder[A],
-    encoder2: TypedEncoder[Out])
-  : TypedAggregateAndColumn[T, A, Out] = {
-    val zeroExpr = Literal.create(summable.zero, encoder2.targetDataType)
+    oencoder: TypedEncoder[Out]
+  ): TypedAggregateAndColumn[T, Out, Out] = {
+    val zeroExpr = Literal.create(summable.zero, TypedEncoder[Out].targetDataType)
     val sumExpr = expr(untyped.sum(column.untyped))
     val sumOrZero = Coalesce(Seq(sumExpr, zeroExpr))
 
-    new TypedAggregateAndColumn[T, A, Out](sumOrZero)
+    new TypedAggregateAndColumn[T, Out, Out](sumOrZero)
   }
 
   def avg[A, T, Out](column: TypedColumn[T, A])(
     implicit
     averageable: CatalystAverageable[A, Out],
-    encoder1: TypedEncoder[A],
-    encoder2: TypedEncoder[Option[Out]]
-  ): TypedAggregateAndColumn[T, A, Option[Out]] = {
-    new TypedAggregateAndColumn[T, A, Option[Out]](untyped.avg(column.untyped))
+    oencoder: TypedEncoder[Out]
```

Comment:
implicit wouldn't be resolved if we replace this with `Out : TypedEncoder`, looks there is a difference because of the order of implicit parameters